tracer/src/Datadog.Trace/Configuration/ExporterSettings.cs

```diff
@@ -177,142 +199,185 @@ public int PartialFlushMinSpans
         /// </summary>
         internal MetricsTransportType MetricsTransport { get; set; }
 
-        private void ConfigureMetricsTransport(IConfigurationSource source, bool forceMetricsOverUdp)
+        internal List<string> ValidationWarnings { get; }
+
+        private void ConfigureMetricsTransport(string agentHost, int dogStatsdPort, string metricsPipeName, string metricsUnixDomainSocketPath)
         {
-            MetricsTransportType? metricsTransport = null;
+            // Agent port is set to zero in places like AAS where it's needed to prevent port conflict
+            // The agent will fail to start if it can not bind a port, so we need to override 8126 to prevent port conflict
+            // Port 0 means it will pick some random available port
+            if (dogStatsdPort < 0)
+            {
+                ValidationWarnings.Add("The provided dogStatsD port isn't valid, it should be positive.");
+            }
+
+            if (dogStatsdPort > 0 || agentHost != null)
+            {
+                // No need to set AgentHost, it is taken from the AgentUri and set in ConfigureTrace
+                MetricsTransport = MetricsTransportType.UDP;
+                DogStatsdPort = dogStatsdPort > 0 ? dogStatsdPort : DefaultDogstatsdPort;
+            }
+            else if (!string.IsNullOrWhiteSpace(metricsPipeName))
+            {
+                MetricsTransport = MetricsTransportType.NamedPipe;
+                MetricsPipeName = metricsPipeName;
+            }
+            else if (metricsUnixDomainSocketPath != null)
+            {
+                SetUdsAsMetricsTransportAndCheckFile(metricsUnixDomainSocketPath, ConfigurationKeys.MetricsUnixDomainSocketPath);
```

Comment:
I think it would improve readability to inline that function here

Reply:
Fixed in ddb553a2a9d31b15b72ff19eec7588818af9171e